iib/web/models.py

```diff
@@ -309,8 +340,21 @@ def _state_to_json(state):
     @classmethod
     def from_json(cls, kwargs):
         # Validate all required parameters are present
-        required_params = {'bundles', 'binary_image'}
+        required_params = {'bundles', 'binary_image', 'operators'}
         optional_params = {'from_index', 'add_arches', 'cnr_token'}
+
+        if 'bundles' in kwargs:
+            request_type = RequestTypeMapping.__members__['add'].value
+            required_params.remove('operators')
```

Comment:
My preference is to add `bundles` to `required_params` rather than remove `operators`. It makes it more intuitive to read.

Reply:
I did think of it but then we would be wrongly representing the two sets in the beginning. I found this approach to be more clear.